pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/ArgumentPatternMatcherTest.java

```diff
@@ -0,0 +1,36 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ArgumentPatternMatcherTest {
+
+    @Test
+    public void should_extract_arguments_if_number() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("I have (\\d+) cukes in my belly"), "I have 36 cukes in my belly");
+        assertThat(arguments).containsExactly("36");
+    }
+
+    @Test
+    public void should_extract_arguments_if_list() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("The following animals: (.*)"), "The following animals: dog, cat, mouse");
+        assertThat(arguments).containsExactly("dog, cat, mouse");
```

Comment:
Shouldn't this be 3 strings instead of one with commas in `containsExactly` check? I'm not sure how should I read this test. Is one argument containing all the content of the given group? My understanding of this feature is that you would like to invoke test method 3 times with `dog`, `cat` and `mouse` separate. Is it this string `"dog, cat, mouse"` parsed later on somewhere?